src/commands/createNewProject/PythonProjectCreator.ts

```diff
@@ -0,0 +1,164 @@
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as path from 'path';
+import * as semver from 'semver';
+import { MessageItem, OpenDialogOptions, workspace } from 'vscode';
+import { UserCancelledError } from 'vscode-azureextensionui';
+import { Platform, PythonAlias, TemplateFilter } from "../../constants";
+import { ext } from '../../extensionVariables';
+import { validateFuncCoreToolsInstalled } from '../../funcCoreTools/validateFuncCoreToolsInstalled';
+import { localize } from "../../localize";
+import { getFuncExtensionSetting, updateWorkspaceSetting } from '../../ProjectSettings';
+import { cpUtils } from "../../utils/cpUtils";
+import { funcHostProblemMatcher, funcHostTaskId } from "./IProjectCreator";
+import { ScriptProjectCreatorBase } from './ScriptProjectCreatorBase';
+
+export class PythonProjectCreator extends ScriptProjectCreatorBase {
+    public readonly templateFilter: TemplateFilter = TemplateFilter.Verified;
+    private readonly pythonVenvPathSetting: string = 'python.venvRelativePath';
+    private pythonAlias: string;
+    public getLaunchJson(): {} {
+        // currently the Python extension launches Windows with type: python and MacOS with type: pythonExperimental
+        const launchType: string = process.platform === Platform.Windows ? 'python' : 'pythonExperimental';
+        return {
+            version: '0.2.0',
+            configurations: [
+                {
+                    name: localize('azFunc.attachToJavaScriptFunc', 'Attach to Python Functions'),
+                    type: launchType,
+                    request: 'attach',
+                    port: 9091,
+                    host: 'localhost',
+                    preLaunchTask: funcHostTaskId
+                }
+            ]
+        };
+    }
+
+    public async addNonVSCodeFiles(): Promise<void> {
+        const funcCoreRequired: string = localize('funcCoreRequired', 'You must have the Azure Functions Core Tools installed to create, debug, and local Python functions.');
+        if (!await validateFuncCoreToolsInstalled(true /* forcePrompt */, funcCoreRequired)) {
+            throw new UserCancelledError();
+        }
+        await this.validatePythonVersion();
+        await this.setVirtualEnviornment();
+        await this.createPythonProject();
+    }
+
+    public async getTasksJson(): Promise<{}> {
+        return {
+            version: '2.0.0',
+            tasks: [
+                {
+                    label: localize('azFunc.runFuncHost', 'Run Functions Host'),
+                    identifier: funcHostTaskId,
+                    type: 'shell',
+                    dependsOn: 'build',
+                    osx: {
+                        command: `source ${await this.getVenvActivatePath(Platform.MacOS)} && func start host`
```

Comment:
See the variable substitution section here: https://code.visualstudio.com/docs/editor/tasks#_variable-substitution

I believe you can do `${config:azureFunctions.python.venvRelativePath}` rather than hard-coding the path

Reply:
So this is kind of tricky because the path changes based on the OS.  The `azureFunctions.python.venvRelativePath` setting is only for locating the root of the virtual environment.  The `getVenvActivatePath` then uses the root to output the path of the activate script based on which OS is passed in.